lib/utils/payment_hash_poller.dart

```diff
@@ -0,0 +1,63 @@
+// ignore_for_file: avoid_print
+import 'dart:async';
+
+import 'package:breez_sdk/bridge_generated.dart';
+import 'package:c_breez/services/injector.dart';
+
+class PaymentHashPoller {
+  final String paymentHash;
+  void Function(bool contains) onDone;
+
+  PaymentHashPoller({
+    required this.onDone,
+    required this.paymentHash,
+  });
+
+  void start() {
+    const int interval = 5;
+    print("Start polling for payment hash: $paymentHash every $interval seconds");
+    final injector = ServiceInjector();
+    final breezLib = injector.breezLib;
+
+    Timer? paymentReceivedTimer;
+
+    paymentReceivedTimer = Timer.periodic(
+      const Duration(seconds: interval),
+      (timer) async {
+        final List<Payment> paymentList = await breezLib.listPayments(
+          filter: PaymentTypeFilter.Received,
+          fromTimestamp: DateTime.now().subtract(const Duration(minutes: 30)).millisecondsSinceEpoch,
+        );
+        for (var payment in paymentList) {
+          final detailsData = payment.details.data;
+          if (!payment.pending && detailsData is LnPaymentDetails && detailsData.paymentHash == paymentHash) {
+            onDone(true);
+            print("Payment received! Stop polling.");
+            paymentReceivedTimer!.cancel();
+          }
+        }
+        print("Payment isn't received yet. Keep polling.");
+      },
+    );
+    // Cancel payment received timer if it's still running after a minute
+    Future.delayed(const Duration(seconds: 60), () => paymentReceivedTimer?.cancel());
```

Comment:
@erdemyerebasmaz  We have this 60 seconds limit both here and higher level of the stack.
Perhaps if we change this method (start) to receive a parameter for the timeout in seconds and also make it return a future (that throws on error or timeout) it will be more self contained.